horizon/enforcer/api.py

```diff
@@ -295,7 +295,14 @@ async def conditional_is_allowed(
         raw_result = json.loads(response.body).get("result", {})
         log_query_result(query, response)
         if legacy_parse_func:
-            raw_result = legacy_parse_func(raw_result)
+            try:
+                raw_result = legacy_parse_func(raw_result)
+            except Exception as e:
```

Comment:
Shouldn't this catch only TypeError?

Reply:
Until now I only raised TypeError inside, but there is a chance someone would want to raise some other exception there.
I don't want the exception to be raised here, I prefer the response of this function to be "invalid" and the router to convert it into the "fallback response" of its' specific type